lib/rucio/core/replica.py

```diff
@@ -919,32 +922,30 @@ def _list_replicas(replicas, show_pfns, schemes, files_wo_replica, client_locati
                 continue
 
             # It's the first time we see this RSE, initialize the protocols needed for PFN generation
-            if rse_id not in protocols_by_rse_id:
+            protocols = protocols_cache.get(rse_id, {}).get(is_archive)
```

Comment:
I don't understand this particular piece. It calls .get() for is_archive which is a bool. what am I missing?